### ISSUE TITLE
Picking the latest version fails for cmip.py builder

### ISSUE THREAD

**Issue by jbusecke** [NCAR/intake-esm-datastore#69]
I have built a catalog file using `cmip.py` with the `--pick-latest-version` flag.
I am still seeing duplicate versions in my catalog file. 

I experimented with the [`_pick_latest_version`](https://github.com/NCAR/intake-esm-datastore/blob/6cd2a369a1cd15b97f0684fe4db66e3ae600c3c9/builders/cmip.py#L103-L124)

locally and was able to rectify my dataframe by adding the field `'dcpp_init_year'` to the fields that are excluded from the groupby call:
```
grpby = list(set(df.columns.tolist()) - {'path', 'version', 'dcpp_init_year'})
```
In my case the `dcpp_init_year` is populated by nans, which might throw of pandas groupby.

**Comment by andersy005** [comment]
Thank you for posting this, @jbusecke! 

> In my case the dcpp_init_year is populated by nans, which might throw of pandas groupby.

This is one thing I overlooked in my previous implementation.

**Comment by andersy005** [comment]
I will update the script

**Comment by jbusecke** [comment]
Would be happy to submit a PR, but prob need some help constructing the tests.

**Comment by andersy005** [comment]
As an update, I addressed this issue in #71.

> Would be happy to submit a PR, but prob need some help constructing the tests.

Constructing tests for the existing utilities is a reasonable thing to do.  It just hasn't been done yet. It would nice to have some tests in the future.

**Comment by jbusecke** [comment]
I just tested the new PR and can confirm that it works. Thanks.
